File: backend/src/main/java/com/ecommerce/backend/service/FavoriteService.java
package com.ecommerce.backend.service;

import com.ecommerce.backend.dto.request.AddFavoriteRequest;
import com.ecommerce.backend.dto.response.FavoriteResponse;

import java.util.List;

public interface FavoriteService {

    FavoriteResponse addFavorite(String tenantName,
                                 Long userId,
                                 AddFavoriteRequest request);

    List<FavoriteResponse> getFavoritesByUser(String tenantName,
                                              Long userId);

    void removeFavorite(String tenantName,
                        Long favoriteId);
}

File: frontend/src/services/favoriteService.js
import api from "./api";

export const addFavorite = async (
  tenantName,
  userId,
  productId
) => {
  return await api.post(
    `/favorites/${tenantName}/user/${userId}`,
    {
      productId,
    }
  );
};

export const getFavorites = async (
  tenantName,
  userId
) => {
  return await api.get(
    `/favorites/${tenantName}/user/${userId}`
  );
};

export const removeFavorite = async (
  tenantName,
  favoriteId
) => {
  return await api.delete(
    `/favorites/${tenantName}/${favoriteId}`
  );
};